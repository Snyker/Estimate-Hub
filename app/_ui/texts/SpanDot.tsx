import React, {HTMLProps, PropsWithChildren} from "react";

export const SpanDot: React.FC<PropsWithChildren<SpanDotProps>> = (
    {
        children,
        dotClass = 'dot',
        isValid = false,
        dotValidClass = 'dot-success',
        validClass = 'text-success',
        className = ''
    }) => {
    return (
        <p className={`text-xs text-gray-11 w-full ${className} ${isValid && validClass}`}><span className={`mr-2 ${dotClass} ${isValid && dotValidClass}`} />{children}</p>
    );
};

export type SpanDotProps = HTMLProps<HTMLParagraphElement> & {
    dotClass?: string;
    isValid?: boolean;
    dotValidClass?: string;
    validClass?: string;
}
