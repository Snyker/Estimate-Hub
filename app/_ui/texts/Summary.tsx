import React, {HTMLAttributes, PropsWithChildren} from "react";

export const Summary: React.FC<PropsWithChildren<HTMLAttributes<HTMLParagraphElement>>> = ({ className = '', children, ...props}) => {
    return (
        <p className={`text-content2 text-sm text-ellipsis overflow-hidden line-clamp-3 ${className}`} {...props}>
            {children}
        </p>
    );
};
