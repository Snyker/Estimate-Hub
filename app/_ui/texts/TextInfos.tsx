import React, {HTMLAttributes} from "react";
import {MdCategory} from "react-icons/md";
import {FaHourglass, FaMoneyBill1Wave} from "react-icons/fa6";
import {FaListAlt} from "react-icons/fa";

export const TextInfos: React.FC<TextInfosProps> = ({ className = '', icon, quantity=0, text, ...props}) => {
    return (
        <div className={`flex flex-row gap-1.5 text-sm text-content3 items-center ${className}`} {...props}>
            { icon && icon}
            { text && <span>{text}</span>}
            { quantity && <span>{quantity}</span>}
        </div>
    );
};

export type TextInfosProps = HTMLAttributes<HTMLDivElement> & {
    icon?: React.ReactElement;
    quantity?: string | number;
    text?: string;
}

export type QuantityInfosProps = Pick<TextInfosProps, "quantity">;

export const TaskInfos: React.FC<QuantityInfosProps> = (props) => {
    return (
        <TextInfos icon={<FaListAlt />} text={"Tâches"} {...props} />
    );
};

export const FeatureInfos: React.FC<QuantityInfosProps> = (props) => {
    return (
        <TextInfos icon={<MdCategory />} text={"Fonctionnalitées"} {...props} />
    );
};

export const CostInfos: React.FC<QuantityInfosProps> = (props) => {
    return (
        <TextInfos icon={<FaMoneyBill1Wave />} text={"Coûts"} {...props} />
    );
};

export const TimeInfos: React.FC<QuantityInfosProps> = (props) => {
    return (
        <TextInfos icon={<FaHourglass />} text={"Temps estimé"} {...props} />
    );
};
