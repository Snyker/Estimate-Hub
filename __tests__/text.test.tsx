import React from 'react';
import renderer from 'react-test-renderer';
import {SpanDot, SpanDotProps} from "@/app/_ui/texts/SpanDot";

describe('SpanDot Component', () => {
    const defaultProps: SpanDotProps = {
        children: 'Some text',
    };

    it('renders correctly with default props', () => {
        const tree = renderer.create(<SpanDot {...defaultProps} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies custom class', () => {
        const tree = renderer.create(<SpanDot {...defaultProps} className="custom-class" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies dot class and text class when isValid is true', () => {
        const tree = renderer.create(<SpanDot {...defaultProps} isValid={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies custom dot class and text class when isValid is true', () => {
        const tree = renderer.create(<SpanDot {...defaultProps} isValid={true} dotValidClass="custom-dot-class" validClass="custom-valid-class" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
