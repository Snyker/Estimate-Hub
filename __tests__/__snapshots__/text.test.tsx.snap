// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpanDot Component applies custom class 1`] = `
<p
  className="text-xs text-gray-11 w-full custom-class false"
>
  <span
    className="mr-2 dot false"
  />
  Some text
</p>
`;

exports[`SpanDot Component applies custom dot class and text class when isValid is true 1`] = `
<p
  className="text-xs text-gray-11 w-full  custom-valid-class"
>
  <span
    className="mr-2 dot custom-dot-class"
  />
  Some text
</p>
`;

exports[`SpanDot Component applies dot class and text class when isValid is true 1`] = `
<p
  className="text-xs text-gray-11 w-full  text-success"
>
  <span
    className="mr-2 dot dot-success"
  />
  Some text
</p>
`;

exports[`SpanDot Component renders correctly with default props 1`] = `
<p
  className="text-xs text-gray-11 w-full  false"
>
  <span
    className="mr-2 dot false"
  />
  Some text
</p>
`;
